### PR TITLE
doc/md/guides: golang migrate diff guide

### PR DESCRIPTION
**Pull request by rotemtam**

### CODE REVIEW

File: doc/md/guides/migration-tools/golang-migrate.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
+---
+id: golang-migrate
+title: Automatic migration planning for golang-migrate
+slug: /guides/migration-tools/golang-migrate
+---
+
+## tl;dr
```
Comment (masseelch) [inline doc/md/guides/migration-tools/golang-migrate.md:7]:
```suggestion
## TLDR
```
Or `TL;DR`

File: doc/md/guides/migration-tools/golang-migrate.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+in your current working directory:
+
+To generate this file run:
+
+```text
+atlas migrate hash --dir file://migrations
+```
+
+Observe a new file named `atlas.sum` was created in your migrations directory
+which contains a hash sum of each file in your directory. For example:
```
Comment (masseelch) [inline doc/md/guides/migration-tools/golang-migrate.md:67]: as well as a total sum. That is the most important part.

**Review by masseelch — COMMENTED** [review]
What I miss in the guide is the possibility to create a migration file without the need for the HCL docment. 

If I have some database I use locally to develop, and I have my migrations dir, I can simply run `atlas migrate diff --to LOCAL_DATABASE_URL` and don't need the way over the HCL representation.

File: doc/md/guides/migration-tools/golang-migrate.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+via an Atlas schema [HCL file](https://atlasgo.io/atlas-schema/sql-resources) or as a
+connection string to a database that contains the desired schema.
+
+In this guide, we will show how Atlas can automatically plan schema migrations for
+golang-migrate users. 
+
+## Pre-requeisites
+
+* An existing project with a `golang-migrate` migrations directory.
+* Docker
```
[the remark this answers is not in the document]
Comment (rotemtam) [reply doc/md/guides/migration-tools/golang-migrate.md:32]: its necessary to follow the examples here

**Comment by a8m** [comment]
> What I miss in the guide is the possibility to create a migration file without the need for the HCL docment.
> 
> If I have some database I use locally to develop, and I have my migrations dir, I can simply run `atlas migrate diff --to LOCAL_DATABASE_URL` and don't need the way over the HCL representation.

Agree with this as well. Looks good besides this.

File: doc/md/guides/migration-tools/golang-migrate.md [hunk 4]
```diff
@@ -0,0 +1,218 @@
+---
+id: golang-migrate
+title: Automatic migration planning for golang-migrate
+slug: /guides/migration-tools/golang-migrate
+---
```
Comment (masseelch) [inline doc/md/guides/migration-tools/golang-migrate.md:5]: do not forget share image

File: doc/md/guides/migration-tools/golang-migrate.md [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+
+## Dev database
+To plan a migration from the current to the desired state, Atlas uses a [Dev Database](/concepts/dev-database),
+which is usually provided by a locally running container with an empty database of the type
+you work with (such as MySQL or PostgreSQL). 
+
+To spin up a local MySQL database that will be used as a dev-database in our example, run:
+
+```text
+docker run --rm --name atlas-db-dev -d -p 3306:3306 -e MYSQL_DATABASE=dev -e MYSQL_ROOT_PASSWORD=pass  mysql:8
```
Comment (masseelch) [inline doc/md/guides/migration-tools/golang-migrate.md:43]:
```suggestion
docker run --rm --name atlas-db-dev -d -p 3306:3306 -e MYSQL_DATABASE=dev -e MYSQL_ROOT_PASSWORD=pass mysql:8
```

File: doc/md/guides/migration-tools/golang-migrate.md [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+```
+
+As reference for the next steps, the URL for the dev-database will be:
```
Comment (masseelch) [inline doc/md/guides/migration-tools/golang-migrate.md:46]:
```suggestion
As reference for the next steps, the URL for the Dev Database will be:
```
You call it `Dev Database` before, now it is `dev-database`. We should use the same notation every time.

File: doc/md/guides/migration-tools/golang-migrate.md [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+Observe a new file named `atlas.sum` was created in your migrations directory
+which contains a hash sum of each file in your directory as well as a total sum.
+For example:
+```text
+h1:y6Zf8kAu98N0jAR+yemZ7zT91nUyECLWzxxR7GHJIAg=
+1_init.down.sql h1:0zpQpoUZcacEatOD+DYXgYD1XvfWUC7EM+agXIRzKRU=
+1_init.up.sql h1:kOM+4u8UsYvvjQMFYAo2hDv5rbx3Mdbh9GvhmbpS0Ig=
+```
+
+### Convert you migrations directory to an Atlas schema file
```
Comment (masseelch) [inline doc/md/guides/migration-tools/golang-migrate.md]:
```suggestion
### Convert your migrations directory to an Atlas schema file
```

File: doc/md/guides/migration-tools/golang-migrate.md [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+  }
+}
+```
+
+Now, let's use Atlas's `migrate diff` command to plan a migration from the current state
+as it exists in the migrations directory to the desired state that is defined by the `schema.hcl`
+file:
+
+```text
+atlas migrate diff --dir file://migrations  --dev-url mysql://root:pass@localhost:3306/dev --to file://schema.hcl --dir-format golang-migrate add_blog_posts
```
Comment (masseelch) [inline doc/md/guides/migration-tools/golang-migrate.md:171]:
```suggestion
atlas migrate diff --dir file://migrations --dev-url mysql://root:pass@localhost:3306/dev --to file://schema.hcl --dir-format golang-migrate add_blog_posts 
```

File: doc/md/guides/migration-tools/golang-migrate.md [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+-- create "blog_posts" table
+CREATE TABLE `blog_posts` (`id` int NOT NULL, `title` varchar(255) NOT NULL, `body` text NOT NULL, `author_id` int NOT NULL, INDEX `blog_author_fk` (`author_id`), CONSTRAINT `blog_author_fk` FOREIGN KEY (`author_id`) REFERENCES `users` (`id`) ON UPDATE NO ACTION ON DELETE NO ACTION) CHARSET utf8mb4 COLLATE utf8mb4_0900_ai_ci;
+```
+And a down migration:
+```sql
+-- reverse: create "blog_posts" table
+DROP TABLE `blog_posts`;
+```
+
+### Alternative: use an existing database as the desired state
```
Comment (masseelch) [inline doc/md/guides/migration-tools/golang-migrate.md]: As a reader, if I see the part about the HCL and don't wanna learn it, I would stop reading. Maybe add a small info box linking here at that place? If someone starts to work with Atlas, they will later adapt HCL, I am sure of it.

File: doc/md/guides/migration-tools/golang-migrate.md [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+
+In some cases, it is convenient to use the schema of an existing database as the desired
+state for your project, instead of defining it in HCL. Atlas's `migrate diff` command can
+plan a migration from your current migration directory sum to a schema. Suppose such a
```
Comment (masseelch) [inline doc/md/guides/migration-tools/golang-migrate.md:202]:
```suggestion
plan a migration from your current migration directory state to an existing schema. Suppose such a 
```

**Pull request merged**